src/v2/components/PotentialFireList.jsx

```diff
@@ -154,13 +223,17 @@ export default function PotentialFireList(props) {
     return tidy
   }, [handlePotentialFire])
 
-  return <FireList
-    fires={fires}
-    firesByKey={firesByKeyRef.current}
-    indexOfOldFires={includesAllFires ? indexOfOldFires : -1}
-    nOldFires={indexOfOldFires > -1 ? allFiresRef.current.length - indexOfOldFires : 0}
-    onToggleAllFires={handleToggleAllFires}
-    region={region}
-    updateFires={updateFires}
-    {...props}/>
+  return 0,
+  <>
+    { shouldNotify && notification && <Notification disableActiveWindow {...notification}/> }
```

Comment:
This is not triggering in my local tests.  shouldNotify and notification are both set, but I don't see notification unless I make a dummy change to this file and save it triggering react dev mode to notice and re-render.

Reply:
Hrmm… I had issues getting this to trigger during development as well. I figured it’s because I’m not running the python detection code in test mode locally. Which is one of the reasons I had the wrong `>` vs `<`—I had commented out the gating code to get the notification to trigger.

I will test with `setInterval()` and see what I find.

Reply:
Okay, I wasn’t able to trigger a notification in an incognito Chrome tab, even with the following hardcoded:

```jsx
  return 0,
  <>
    <Notification title="HELLO"/>
    ...
  </>
```

So I think this is a regression in Chrome? I was able to trigger an incognito notification before I submitted the PR, though I did have to whitelist `http://localhost:3000` in order to do so. Now, whether I whitelist it or not, I don’t see any notifications in incognito. A normal window works as expected.

Unfortunately, it looks like `disableActiveWindow` is also fubar. If it’s set (to `true`), I never see a notification. If it’s not set or `false`, everything works. I’ve tested with Chrome, Firefox, and Safari with normal and incognito tabs… Weird.

I’m also experiencing weird issues with macOS, maybe? Chrome windows appear to have focus while other windows are on top. And Chrome won’t quit properly. Going to reboot to see if there’s a difference.

Reply:
Incognito disables notifications for my setup, so I test notifications without incognito.

disableActiveWindow disables notification if the browser tab/window is in the foreground.  So, I test with a command shell window in the foreground that partly covers up the browser window.

Reply:
> disableActiveWindow disables notification if the browser tab/window is in the foreground. So, I test with a command shell window in the foreground that partly covers up the browser window.

Unfortunately, I already knew this and was testing accordingly. If `disableActiveWindow` is `true`, I never see a notification no matter how the windows are arranged. Which may explain why you’re not seeing the notification in your local testing.

Reply:
![screencast 2022-09-13 15-52-31](https://user-images.githubusercontent.com/377413/190007126-86a2276e-10b8-4cef-b635-a33e18b2f686.gif)

Reply:
Are you sure it's not a react issue?  I see the notifications as soon as I save JSX file after adding a "space".

Reply:
I am definitely unable to trigger notifications with `disableActiveWindow === true`. Whether the Chrome tab has focus or not. I get inconsistent behavior if I touch the code to trigger a React dev re-render when `disableActiveWindow` is `true`.

Reply:
Strange...  Maybe try v1 with setInterval to see how that behaves on your system?

Reply:
v1 also does not work with `disableActiveWindow === true`

Reply:
Here’s the problem… The `Notification` component isn’t constructed and it’s `state` isn’t set until the first time it renders. The first time it renders, it assumes the window is focused. In my testing, I’m reloading and then blurring the window. But since the `Notification` component has never rendered, it’s never been initialized and when it does render because there’s a notification to show, it thinks the window is focused. This is all ultimately due to the fact that the component is manually tracking focus state with `focus`/`blur` events. In addition to fixing this upstream by preferring the `document.hasFocus()` method and only falling back to manual event management if `hasFocus()` isn’t supported, we can probably fix this by rendering notification without any props? Checking now.

Reply:
Fixed locally to force the Notification component to initialize. Will submit an upstream fix shortly.

Reply:
https://github.com/mobilusoss/react-web-notification/pull/98

Reply:
The new behavior on my testing is that the notification show up sometimes. disableActiveWindow seems flaky in v2 but reliable in v1.  Will your hasFocus change fix that?  Other than that, the PR looks great.

Reply:
Hmm… I’m not sure that my upstream hasFocus fix will correct the flakiness that you’re seeing… In my testing, both forcing instantiation and using hasFocus in the Notification implementation produce rock-solid results. That you’re experiencing flakiness worries me. To make sure that we’re testing the same thing, can you add the following to `PotentialFireList.jsx` before the render?

```js
  const timerRef = useRef()

  useEffect(() => {
    clearTimeout(timerRef.current)
    timerRef.current = setTimeout(() => {
      const clone = {...allFiresRef.current[0]}
      clone.cameraID = 'xyz-test-camera'
      clone.timestamp = clone.sortId = Math.round(Date.now() / 1000) - 30
      clone.isRealTime = true

      console.log('timeout')
      handlePotentialFire({data: JSON.stringify(clone)})
    }, 6000)
  }, [handlePotentialFire])
```

And maybe share what steps you’re taking to reproduce the flaky behavior?

Reply:
Now specifying default props (including `disableActiveWindow`) in JSX instead of from  an object spread.